### PR TITLE
Hrobbins/pagination component 174

### PR DESCRIPTION
**Pull request by pr3tori4n**
Reference Issue: #174 

* This PR adds the pagination component, including styles, tests for the component, and stories for the component.
* I've added some utility code to enable a11y testing using puppeteer like we have in our app-components repo.
* I've setup the demo as a separate HTML file with a link to it from index and modified the build to copy that directory.

### CODE REVIEW

File: src/components/calcite-pagination/calcite-pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        this.selectedIndex = num;
+      }}>{num}</a>
+    );
+  }
+
+  renderLeftEllipsis() {
+    if ( this.total > maxPagesDisplayed && this.showLeftEllipsis() ) {
+      return (
+        <span class={`${CSS.ellipsis} ${CSS.ellipsisStart}`}>
+          <CalciteIcon size="16" path={ellipsis16} />
```
Comment (macandcheese) [inline src/components/calcite-pagination/calcite-pagination.tsx]: Could probably use the new `calcite-icon` component here.

File: src/components/calcite-pagination/calcite-pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        </span>
+      );
+    }
+  }
+
+  renderRightEllipsis() {
+    if ( this.total > maxPagesDisplayed && this.showRightEllipsis() ) {
+      return (
+        <span class={`${CSS.ellipsis} ${CSS.ellipsisEnd}`}>
+          <CalciteIcon size="16" path={ellipsis16} />
```
Comment (macandcheese) [inline src/components/calcite-pagination/calcite-pagination.tsx]: Same as above

File: src/components/calcite-pagination/calcite-pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  Method,
+  State,
+  Watch
+} from "@stencil/core";
+import { getElementDir } from "../../utils/dom";
+import { CSS, TEXT } from "./resources";
+import { chevronLeft16, chevronRight16, ellipsis16 } from "@esri/calcite-ui-icons";
+import CalciteIcon from "../../utils/CalciteIcon";
+
+const maxPagesDisplayed = 5;
```
Comment (macandcheese) [inline src/components/calcite-pagination/calcite-pagination.tsx]: Could this be user configurable at some point?

File: src/components/calcite-pagination/calcite-pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        }
+      }
+    }
+
+    while (currentNum <= end) {
+      pages.push(currentNum);
+      currentNum ++;
+    }
+
+    return pages.map(page => {
```
Comment (paulcpederson) [inline src/components/calcite-pagination/calcite-pagination.tsx]: you could just online this as it has a single line return: `return pages.map(page => this.renderPage(page));`

**Comment by paulcpederson** [comment]
Looks good :+1:

File: src/components/calcite-pagination/calcite-pagination.tsx (hunk 3 above)
Comment (pr3tori4n) [reply src/components/calcite-pagination/calcite-pagination.tsx]: Yea I think this could be a future enhancement.

**Pull request merged**